packages/application-generic/src/health/queue-health-indicator.service.ts

```diff
@@ -0,0 +1,67 @@
+import {
+  HealthCheckError,
+  HealthIndicator,
+  HealthIndicatorResult,
+} from '@nestjs/terminus';
+import { Injectable, Logger } from '@nestjs/common';
+
+import { QueueBaseService } from '../services';
+import { IHealthIndicator } from './health-indicator.interface';
+
+@Injectable()
+export abstract class QueueHealthIndicator
+  extends HealthIndicator
+  implements IHealthIndicator
+{
+  constructor(
+    private queueBaseService: QueueBaseService,
+    private indicatorKey: string,
+    private serviceName: string,
+    private logContext: string
+  ) {
+    super();
+  }
+
+  async isHealthy(): Promise<HealthIndicatorResult> {
+    return await this.handleHealthCheck();
+  }
+
+  async isActive(): Promise<HealthIndicatorResult> {
+    return await this.handleHealthCheck(true);
+  }
+
+  async handleHealthCheck(checkActive = false) {
+    const isReady = this.queueBaseService.isReady();
+
+    if (!isReady) {
+      Logger.verbose(`${this.serviceName} is not ready`, this.logContext);
+
+      throw new HealthCheckError(
+        `${this.serviceName} Health`,
+        this.getStatus(this.indicatorKey, false)
+      );
+    }
+
+    if (!checkActive) {
+      Logger.verbose(`${this.serviceName} is ready`, this.logContext);
+
+      return this.getStatus(this.indicatorKey, true);
+    }
+
+    const isPaused = await this.queueBaseService.isPaused();
+    const isActive = isReady && !isPaused;
+
+    if (!isActive) {
+      Logger.verbose(`${this.serviceName} is not active`, this.logContext);
+
+      throw new HealthCheckError(
+        `${this.serviceName} Health`,
+        this.getStatus(this.indicatorKey, false)
+      );
+    }
+
+    Logger.verbose(`${this.serviceName} is active`, this.logContext);
+
+    return this.getStatus(this.indicatorKey, true);
+  }
```

Comment:
I think we can avoid the use of argument flags and with a single function compose both methods in a more maintainable way. What do you think?

```suggestion
  async isHealthy(): Promise<HealthIndicatorResult> {
    const isReady = this.checkIsReady();
     
    Logger.verbose(`${this.serviceName} is ready`, this.logContext);

    return this.getStatus(this.indicatorKey, true);
  }

  async isActive(): Promise<HealthIndicatorResult> {
    const isReady = this.checkIsReady();
    const isPaused = await this.queueBaseService.isPaused();
    const isActive = isReady && !isPaused;

    if (!isActive) {
      Logger.verbose(`${this.serviceName} is not active`, this.logContext);

      throw new HealthCheckError(
        `${this.serviceName} Health`,
        this.getStatus(this.indicatorKey, false)
      );
    }

    Logger.verbose(`${this.serviceName} is active`, this.logContext);

    return this.getStatus(this.indicatorKey, true);
  }
  
  private checkIsReady(): boolean {
      const isReady = this.queueBaseService.isReady();

    if (!isReady) {
      Logger.verbose(`${this.serviceName} is not ready`, this.logContext);

      throw new HealthCheckError(
        `${this.serviceName} Health`,
        this.getStatus(this.indicatorKey, false)
      );
    }
    
    return isReady;
  }
```

Reply:
Do you mean that one function will check isHealth and isActive altogether?